spec/ruby/core/binding/eval_spec.rb

```diff
@@ -13,6 +13,16 @@
     bind.eval("Inside.name").should == "BindingSpecs::Demo::Inside"
   end
 
+  it "does not leak variables to cloned bindings" do
+    obj = BindingSpecs::Demo.new(1)
+    bind = obj.get_empty_binding
+    bind2 = bind.dup
+
+    bind.eval("x = 72")
```

Comment:
I think using Binding#local_variable_set would be more resilient here, e.g. if an implementation does not have Binding#eval declare variables in the binding.